Project/cypress/support/pageobjects/Report.ts

```diff
@@ -0,0 +1,74 @@
+export class Report {
+    static elements = {
+        reportBtn: () => cy.get('.oxd-topbar-body-nav > ul',{timeout:40000}).contains("Report"),
+        addBtn: () => cy.get('.orangehrm-header-container > .oxd-button',{timeout:40000}),
+        reportName: () => cy.get('.oxd-input',{timeout:40000}).last(),
+        selectBox: () => cy.get('.oxd-select-text',{timeout:40000}),
+        dropDown: () => cy.get('.oxd-select-dropdown',{timeout:40000}),
+        iconBtn: () => cy.get(' .oxd-icon-button',{timeout:40000}),
+        loadingSpinner: () => cy.get('.oxd-loading-spinner-container',{timeout:40000}),
+        switch: () => cy.get('.oxd-switch-input',{timeout:40000}),
+        saveBtn: () => cy.get('.oxd-button--secondary',{timeout:40000}),
+        searchBtn: () => cy.get('.oxd-form-actions > .oxd-button--secondary',{timeout:40000}),
+        deleteIcon: () => cy.get('.oxd-table-cell-actions > :nth-child(1) > .oxd-icon',{timeout:40000}),
+    }
+    static selectSearchCriteria(job: string, location: string) {
+
+        this.elements.selectBox().first().click({ force: true })
+        this.elements.dropDown().contains('Job Title').click({ force: true })
+        this.elements.iconBtn().eq(2).click({ force: true })
+        this.elements.selectBox().eq(2).click({ force: true })
+        this.elements.dropDown().contains(job,{timeout:40000}).click({ force: true })
+
+
+        this.elements.selectBox().first().click({ force: true })
+        this.elements.dropDown().contains('Location').click({ force: true })
+        this.elements.iconBtn().eq(2).click({ force: true })
+        this.elements.selectBox().eq(3).click({ force: true })
+        this.elements.dropDown().contains(location,{timeout:40000}).click({ force: true })
+    }
```

Comment:
we don't prefer to do a function to fill out whole the form, you should do a function for each action and call them in your test